security/pkg/nodeagent/caclient/providers/vault/client.go

```diff
@@ -19,70 +19,161 @@ import (
 	"crypto/tls"
 	"crypto/x509"
 	"fmt"
+	"io/ioutil"
 	"net/http"
 	"strconv"
+	"strings"
 
-	"github.com/hashicorp/vault/api"
+	vaultapi "github.com/hashicorp/vault/api"
 
-	caClientInterface "istio.io/istio/security/pkg/nodeagent/caclient/interface"
+	"istio.io/istio/security/pkg/util"
+	"istio.io/pkg/env"
 	"istio.io/pkg/log"
 )
 
+const (
+	envVaultAddr        = "VAULT_ADDR"
+	envVaultTLSCertPath = "VAULT_AUTH_CERT_PATH"
+	envJwtPath          = "VAULT_AUTH_JWT_PATH"
+	envLoginRole        = "VAULT_LOGIN_ROLE"
+	envLoginPath        = "VAULT_LOGIN_PATH"
+	envSignCsrPath      = "VAULT_SIGN_CSR_PATH"
+	envCaCertPath       = "VAULT_CA_CERT_GET_PATH"
+
+	certKeyInCACertResp        = "certificate"
+	certKeyInCertSignResp      = "certificate"
+	caChainKeyInCertSignResp   = "ca_chain"
+	issuingCAKeyInCertSignResp = "issuing_ca"
+	jwtKeyInLoginReq           = "jwt"
+	roleKeyInLoginReq          = "role"
+)
+
 var (
 	vaultClientLog = log.RegisterScope("vault", "Vault client debugging", 0)
 )
 
-type vaultClient struct {
-	enableTLS   bool
-	tlsRootCert []byte
-
-	vaultAddr        string
-	vaultLoginRole   string
-	vaultLoginPath   string
-	vaultSignCsrPath string
+// VaultClient is a client for interaction with Vault.
+type VaultClient struct {
+	enableTLS       bool
```

Comment:
I would make those visible - and/or create a VaultOptions struct.

The env variables are meant for experiments/temp flags/short-lived/internal stuff. 
If we want to move this out of 'experimental' - it'll need proper ProxyConfig settings.

Reply:
And I would make it 'CaConfig' - not VaultClient. 

Virtually all external CAs will take similar parameters.